src/bytes_mut.rs

```diff
@@ -838,7 +906,10 @@ impl BytesMut {
             return true;
         }
 
-        self.reserve_inner(additional, false)
+        match self.reserve_inner(additional, false) {
+            Err(err) => panic!("fail to reserve: {}", err),
```

Comment:
This codepath (`try_reclaim` error case in `reserve_inner`) actually can't be reached (reserve_inner returns an error when it fails to allocate, but this codepath calls reserve_inner with `allocate` set to false). It should probably be changed to `unreachable!()`.

For the other caller (`try_reserve` error case), it requires creating a `Layout` to pass to `handle_alloc_error`. I currently have this:

```rust
                let mut layout = Layout::new::<u8>();
                if let Ok(array_layout) = layout.array(self.len() + additional) {
                    layout = array_layout;
                }
                std::alloc::handle_alloc_error(array_layout);
```

Ergo we create a `Layout` of `[u8; len + additional]`, and if that fails, we fallback to a simple layout of `u8`. I think that's good enough? In practice, `[u8; len + additional]` should always work out, but if len + additional overflows isize, it might error out - and I'm not sure what to give to `handle_alloc_error` in this case.